x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/PhysicalPlanOptimizerTests.java

```diff
@@ -1683,6 +1683,24 @@ public void testNoTextFilterPushDown() {
         assertNull(source.query());
     }
 
+    public void testNoNonIndexedFilterPushDown() {
+        var plan = physicalPlan("""
+            from test
+            | where long_noidx == 1
+            """);
+
+        var optimized = optimizedPlan(plan);
+        var limit = as(optimized, LimitExec.class);
+        var exchange = asRemoteExchange(limit.child());
+        var project = as(exchange.child(), ProjectExec.class);
+        var extract = as(project.child(), FieldExtractExec.class);
+        var limit2 = as(extract.child(), LimitExec.class);
+        var filter = as(limit2.child(), FilterExec.class);
+        var extract2 = as(filter.child(), FieldExtractExec.class);
+        var source = source(extract2.child());
+        assertNull(source.query());
```

Comment:
What is this test trying to do? Assert that there is no filter being pushed down to ES for `long_noidx` because the fields itself is not indexed?
If so, shouldn't there be no actual query that gets pushed down, not even some kind of extraction? Meaning `long_noidx` is considered `null` then the `where long_noidx == 1` is always `false`.

Reply:
Yes, in principle we should have one more optimization rule that folds that expression and evaluates to false, but we don't have this logic so far. 
We'll probably need some more information to be accurate on this (for now we are only catching exceptions at extraction time, but some cases will actually work, eg. TEXT is not aggregatable and won't be pushed down, but it will be extracted and the normal filtering will work just fine).
I'll open an issue so that we don't forget